capnp/src/private/layout.rs

```diff
@@ -3479,6 +3479,11 @@ impl<'a> StructReader<'a> {
         }
     }
 
+    #[inline]
+    pub fn is_pointer_field_null(&self, ptr_index: WirePointerCount) -> bool {
```

Comment:
What if `ptr_index` is greater than or equal to `self.pointer_count`? That can happen if the writer of the message is using an old version of the schema that has fewer fields defined.

Reply:
I was assuming that the check had to be somewhere else because otherwise I don't see how the rest of this file works.

I copied the implementation of `StructReader::is_pointer_field_null`.

Doesn't `StructBuilder::get_pointer_field(...).is_null()` also ultimately boil down to the same an unchecked addition?

```rust
impl<'a> StructBuilder<'a> {
    // ...
    pub fn get_pointer_field(self, ptr_index: WirePointerCount) -> PointerBuilder<'a> {
        PointerBuilder {
            arena: self.arena,
            segment_id: self.segment_id,
            cap_table: self.cap_table,
            pointer: unsafe { self.pointers.add(ptr_index) },
        }
    }
    // ...
}

// ...
    
impl<'a> PointerBuilder<'a> {
     // ...
    #[inline]
    pub fn is_null(&self) -> bool {
        unsafe { (*self.pointer).is_null() }
    }
    // ...
}
 ```

Reply:
Builders are guaranteed to have the expected number of pointers. Readers are not.